src/endpoint/scheduler.rs

```diff
@@ -370,6 +374,36 @@ impl JobHandle {
     }
 }
 
+impl Drop for JobHandle {
+    fn drop(&mut self) {
+        debug!("Cleaning up JobHandle");
+        if self.container_id.is_some() {
+            debug!("Container was already started");
+            let docker = self.endpoint.docker().clone();
+            let container_id = self.container_id.take().unwrap();
+
+            tokio::spawn(async move {
+                let container = docker.containers().get(&container_id);
+                let container_info = container.inspect().await.unwrap();
```

Comment:
We should avoid the unwrap here - I'd probably just log the error (in theory we might occasionally run into such errors when the containers terminate between the if and this inspect).

PS: We want to avoid `unwrap()` as much as possible in general (but there are of course exceptions where it's fine).

Reply:
Fe2O3.unwrap().unwrap().unwrap().unwrap()